devito/types/tensor.py

```diff
@@ -31,16 +67,22 @@ class TensorFunction(AbstractTensor, Differentiable):
         Staggering of each component, needs to have the size of the tensor. Defaults
         to the Dimensions.
     """
+    _is_TimeDependent = False
     _sub_type = Function
-    _op_priority = Differentiable._op_priority + 2.
     _class_priority = 10
+    _op_priority = Differentiable._op_priority + 1.
 
     def __init_finalize__(self, *args, **kwargs):
-        self._is_symmetric = kwargs.get('symmetric', True)
-        self._is_diagonal = kwargs.get('diagonal', False)
-        self._staggered = kwargs.get('staggered', self.space_dimensions)
-        self._grid = kwargs.get('grid')
-        self._space_order = kwargs.get('space_order', 1)
+        if args:
```

Comment:
`if len(args) != ... : raise ValueError(...)`

?

Reply:
Not needed, would already have been caugtht by sympy.